pvt/common/setupTests.ts

```diff
@@ -101,28 +101,28 @@ chai.use(function (chai, utils) {
           if (!matches || matches.length !== 6) throw error;
 
           const expectedReason: string = matches[2];
-          let actualErrorCode: string = matches[4];
-
-          // If the actual error code is not a balancer error, re-throw error
-          if (!actualErrorCode.includes('BAL#')) throw error;
-          actualErrorCode = actualErrorCode.replace('BAL#', '');
-
-          // @ts-ignore
-          let actualReason = reasons[actualErrorCode];
-          if (!actualReason) actualReason = 'Could not match a Balancer error message';
+          const actualErrorCode: string = matches[4];
+
+          let actualReason: string;
+          try {
+            actualReason = parseBalancerErrorCode(actualErrorCode);
+          } catch {
+            // If it's not a Balancer error then rethrow
+            if (!error.includes('BAL#')) return error;
+            // Otherwise it might be a Balancer error we don't know about yet.
+            actualReason = 'Could not match a Balancer error message';
+          }
 
-          // If there is no balancer error matching the expected revert reason re-throw the error
-          // @ts-ignore
-          const expectedError = Object.entries(reasons).find(([, value]) => value == expectedReason);
-          if (!expectedError) {
+          try {
+            // Assert the error code matched the actual reason
+            const expectedErrorCode = encodeBalancerErrorCode(expectedReason);
+            const message = `Expected transaction to be reverted with BAL#${expectedErrorCode} (${expectedReason}), but other exception was thrown: Error: VM Exception while processing transaction: revert BAL#${actualErrorCode} (${actualReason})`;
+            expect(actualErrorCode).to.be.equal(expectedErrorCode, message);
```

Comment:
I think this might also be wrong - we don't want to catch the exception that `except` might throw: I think we don't want to put this inside the `try` block.

Reply:
Yeah, I wasn't thinking about the `expect`'s error. I've pulled that out to ensure that throws correctly.